File: apps/web/types/chat-message.ts
export interface ChatMessage {
  id: string
  taskId?: string
  content: string
  role: "user" | "assistant" | "system"
  createdAt: string
  metadata?: Record<string, any>
}

export interface ChatMessageCreateInput {
  taskId?: string
  content: string
  role: "user" | "assistant" | "system"
  metadata?: Record<string, any>
} 

File: apps/mobile-flutter/lib/features/chat/models/chat_message.dart
import 'package:json_annotation/json_annotation.dart';

part 'chat_message.g.dart';

enum MessageRole {
  @JsonValue('user')
  user,
  @JsonValue('assistant')
  assistant,
  @JsonValue('system')
  system,
}

@JsonSerializable()
class ChatMessage {
  final String id;
  final MessageRole role;
  final String content;
  final String? taskId;
  final DateTime createdAt;
  final DateTime updatedAt;

  ChatMessage({
    required this.id,
    required this.role,
    required this.content,
    this.taskId,
    required this.createdAt,
    required this.updatedAt,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json) =>
      _$ChatMessageFromJson(json);
  Map<String, dynamic> toJson() => _$ChatMessageToJson(this);
}
